### PR TITLE
Jt 15 create rest for satellite

### PR DESCRIPTION
**Pull request by saffchen**

### CODE REVIEW

File: stock-system/src/main/java/saffchen/controller/satellite/SatelliteRestController.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+public class SatelliteRestController {
+
+    private final SatelliteService satelliteService;
+
+    @Autowired
+    public SatelliteRestController(SatelliteService satelliteService){
+        this.satelliteService = satelliteService;
+    }
+
+    @GetMapping("/all")
```
Comment (saffchen) [inline stock-system/src/main/java/saffchen/controller/satellite/SatelliteRestController.java:25]: @freesky15, Please follow Best Practice in naming requests

Comment (pelmegov) [inline stock-system/src/main/java/saffchen/controller/satellite/SatelliteRestController.java:25]: Let's remove /all, /id, /name from path. It's not a good practice in rest api.

File: stock-system/src/main/java/saffchen/controller/satellite/SatelliteRestController.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    public List<SatelliteDTO> getAllSatellitesWithProductCount(){
+        return satelliteService.findAll();
```
Comment (saffchen) [inline stock-system/src/main/java/saffchen/controller/satellite/SatelliteRestController.java:27]: @freesky15, Should be returned ResponseEntity

File: stock-system/src/main/java/saffchen/controller/satellite/SatelliteRestController.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    }
+
+    @GetMapping("/id/{id}")
```
Comment (saffchen) [inline stock-system/src/main/java/saffchen/controller/satellite/SatelliteRestController.java:30]: @freesky15, same way naming

File: stock-system/src/main/java/saffchen/entities/ProductEntity.java [hunk 4]
```diff
@@ -1,86 +1,105 @@
...
 import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
 
 @Entity
 @Getter
 @Setter
 @NoArgsConstructor
 @AllArgsConstructor
-@Table(name = "Product")
```
Comment (pelmegov) [inline stock-system/src/main/java/saffchen/entities/ProductEntity.java:18]: Looks like correct formatting has been changed to incorrect one.

File: stock-system/src/main/java/saffchen/controller/satellite/SatelliteRestController.java [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    public SatelliteEntity getSatellite(@PathVariable("id") Long id){
+
+        return satelliteService.findById(id);
+    }
+
+    @GetMapping("/name/{name}")
```
Comment (saffchen) [inline stock-system/src/main/java/saffchen/controller/satellite/SatelliteRestController.java:36]: @freesky15, same way naming

File: stock-system/src/main/java/saffchen/entities/ProductEntity.java [hunk 6]
```diff
@@ -1,86 +1,105 @@
...
+    }
+
+    public String getTags(){
+        return this.tags.stream().collect(Collectors.joining(",")).toString();
     }
 
     @Override
-    public String toString() {
-        return title + ", " + description + ", " + price + ", " + tags + ", " + category + ", " + count + ", " + satellite.toString();
+    public boolean equals(Object o) {
```
Comment (pelmegov) [inline stock-system/src/main/java/saffchen/entities/ProductEntity.java:75]: Lombok can replace some trivial boilerplate.

File: stock-system/src/main/java/saffchen/service/SatelliteService.java [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+        List<SatelliteDTO> sdtos =  satelliteRepository.findAllSatellitesWithProductCount();
+        for (SatelliteDTO sdto: sdtos) {
+            System.out.println(sdto.getId() + " : " + sdto.getName() + " : " + sdto.getDescription() + " : " + sdto.getCount());
+        }
+        return sdtos;
+    }
+    private SatelliteDTO convertToSatelliteDTO(SatelliteEntity satelliteEntity){
+        SatelliteDTO satelliteDTO = new SatelliteDTO();
+        //satelliteDTO.setId(Long.valueOf(satelliteEntity.getId()));
+        satelliteDTO.setName(satelliteEntity.getName());
```
Comment (saffchen) [inline stock-system/src/main/java/saffchen/service/SatelliteService.java:41]: @freesky15, If we do not use this, then the commented out line should be removed

File: stock-system/src/main/java/saffchen/repositories/SatelliteRepository.java [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+import saffchen.entities.SatelliteEntity;
+
+import java.util.List;
+import java.util.Optional;
+
+@Repository
+public interface SatelliteRepository extends JpaRepository<SatelliteEntity, Long> {
+    Optional<SatelliteEntity> findByNameIgnoreCase(String name);
+
+    @Query("SELECT new saffchen.dto.SatelliteDTO(se.id, se.name, se.description, COUNT(p.id)) " +
```
Comment (pelmegov) [inline stock-system/src/main/java/saffchen/repositories/SatelliteRepository.java:16]: It's not a best idea to map entity to the dto right here, let's do it outside of repository/dao layer.

File: stock-system/src/main/java/saffchen/service/SatelliteService.java [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+    }
+
+    public List<SatelliteDTO> findAll(){
+        List<SatelliteDTO> sdtos =  satelliteRepository.findAllSatellitesWithProductCount();
+        for (SatelliteDTO sdto: sdtos) {
+            System.out.println(sdto.getId() + " : " + sdto.getName() + " : " + sdto.getDescription() + " : " + sdto.getCount());
+        }
+        return sdtos;
+    }
+    private SatelliteDTO convertToSatelliteDTO(SatelliteEntity satelliteEntity){
```
Comment (pelmegov) [inline stock-system/src/main/java/saffchen/service/SatelliteService.java:38]: This is not a service level transformation and can be replaced by mapstruct.

File: stock-system/src/main/java/saffchen/service/SatelliteService.java [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+import javax.persistence.PersistenceContext;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+@Transactional(readOnly = true)
+public class SatelliteService {
+    @PersistenceContext
+    private EntityManager em;
```
Comment (pelmegov) [inline stock-system/src/main/java/saffchen/service/SatelliteService.java:22]: I don't think that entity manager is required here.

File: stock-system/src/main/java/saffchen/service/SatelliteService.java [hunk 11]
```diff
@@ -0,0 +1,56 @@
...
+    @Autowired
+    public SatelliteService(SatelliteRepository satelliteRepository) {
+
+        this.satelliteRepository = satelliteRepository;
+    }
+
+    public List<SatelliteDTO> findAll(){
+        List<SatelliteDTO> sdtos =  satelliteRepository.findAllSatellitesWithProductCount();
+        for (SatelliteDTO sdto: sdtos) {
+            System.out.println(sdto.getId() + " : " + sdto.getName() + " : " + sdto.getDescription() + " : " + sdto.getCount());
```
Comment (pelmegov) [inline stock-system/src/main/java/saffchen/service/SatelliteService.java:34]: We shouldn't use printing to console anywhere, let's use logger, but just for the valueable information - for troubleshooting or some critical sections logging.

File: stock-system/src/main/java/saffchen/service/SatelliteService.java [hunk 12]
```diff
@@ -0,0 +1,56 @@
...
+import saffchen.repositories.SatelliteRepository;
+
+import javax.persistence.EntityManager;
+import javax.persistence.PersistenceContext;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+@Transactional(readOnly = true)
```
Comment (pelmegov) [inline stock-system/src/main/java/saffchen/service/SatelliteService.java:19]: Let's move the Transactional annotation to method level. We actually would like to store a new satellites later.

File: stock-system/src/main/java/saffchen/service/SatelliteService.java [hunk 13]
```diff
@@ -0,0 +1,56 @@
...
+    private SatelliteDTO convertToSatelliteDTO(SatelliteEntity satelliteEntity){
+        SatelliteDTO satelliteDTO = new SatelliteDTO();
+        //satelliteDTO.setId(Long.valueOf(satelliteEntity.getId()));
+        satelliteDTO.setName(satelliteEntity.getName());
+        //satelliteDTO.setDescription(satelliteDTO.getDescription());
+        return satelliteDTO;
+    }
+
+    public SatelliteEntity findById(Long id){
+        Long convertedId = Long.valueOf(id);
```
Comment (pelmegov) [inline stock-system/src/main/java/saffchen/service/SatelliteService.java:47]: Why do we need that transformation from Long to Long?

File: stock-system/src/main/java/saffchen/service/SatelliteService.java [hunk 14]
```diff
@@ -0,0 +1,56 @@
...
+        Optional<SatelliteEntity> foundSatellite = satelliteRepository.findById(convertedId);
+        return foundSatellite.orElse(null);
+    }
+
+    public SatelliteEntity findByNameIgnoreCase(String name){
+        Optional<SatelliteEntity> foundSatellite = satelliteRepository.findByNameIgnoreCase(name);
+        return foundSatellite.orElse(null);
```
Comment (pelmegov) [inline stock-system/src/main/java/saffchen/service/SatelliteService.java:54]: Minor: Let's group to method chain.

File: stock-system/src/main/java/saffchen/utils/SatelliteExceptionResponce.java [hunk 15]
```diff
@@ -0,0 +1,6 @@
+package saffchen.utils;
+
+public class SatelliteExceptionResponce {
```
Comment (pelmegov) [inline stock-system/src/main/java/saffchen/utils/SatelliteExceptionResponce.java:3]: Why do we need that class? Let's remove it or transform to exception like SatelliteNotFoundException or something like similar.

**Review by pelmegov — CHANGES_REQUESTED** [review]

File: stock-system/src/main/java/saffchen/service/SatelliteService.java [hunk 16]
```diff
@@ -0,0 +1,56 @@
+package saffchen.service;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.jpa.repository.Query;
```
Comment (saffchen) [inline stock-system/src/main/java/saffchen/service/SatelliteService.java:4]: minor: unused dependencies

**Review by fstk1337 — COMMENTED** [review]
Agreed with existing comments. Directory stock-system should be renamed to stock to be able to merge into dev.

**Pull request closed without merging**